tfsdk/config.go

```diff
@@ -21,10 +21,43 @@ func (c Config) Get(ctx context.Context, target interface{}) diag.Diagnostics {
 	return reflect.Into(ctx, c.Schema.AttributeType(), c.Raw, target, reflect.Options{})
 }
 
-// GetAttribute retrieves the attribute found at `path` and returns it as an
+// GetAttribute retrieves the attribute found at `path` and populates the
+// `target` with the value.
+func (c Config) GetAttribute(ctx context.Context, path *tftypes.AttributePath, target interface{}) diag.Diagnostics {
+	attrValue, diags := c.getAttributeValue(ctx, path)
+
+	if attrValue == nil {
```

Comment:
Would we not want `diags.HasError()` here instead?

Reply:
You're right we should check for error diagnostics, although for safety we'll probably want both checks. I think the reason I chose `attrValue == nil` is because it will otherwise panic in `ValueAs()`... or something... but I forgot to add the error check as well.